### PR TITLE
Create readme file for anonymizer

### PR DESCRIPTION
**Pull request by shiranr**
Create readme.md file for anonymizer and the anonymize endpoint.
Add explanation on how the anonymizer handles different scenarios

### CODE REVIEW

File: presidio-anonymizer/README.MD [hunk 1]
```diff
@@ -1 +1,121 @@
-# TODO add the document Omri wrote.
```
Comment (navalev) [inline presidio-anonymizer/README.MD:1]: add task/user story here for the TODO

Comment (shiranr) [reply presidio-anonymizer/README.MD:1]: This is no longer relevant. I did for others.

File: presidio-anonymizer/README.MD [hunk 2]
```diff
@@ -1 +1,121 @@
...
\ No newline at end of file
+# Presidio anonymizer
+
+## Description
+
+The Presidio anonymizer is a Python based service for anonymizing detected PII text
```
Comment (balteravishay) [inline presidio-anonymizer/README.MD:5]: not sure service is the right word. maybe module?

File: presidio-anonymizer/README.MD [hunk 3]
```diff
@@ -1 +1,121 @@
...
+
+- One PII - uses a given or default transformation to anonymize and replace the PII text
+  entity.
+- Full overlap of PIIs - When one text have several PIIs, the PII with the higher score
+  will be taken. Between PIIs with similar scores, the selection will be arbitrary.
+- One PII is contained in another - anonymizer will use the PII with larger text.
+- Partial intersection - both will be returned concatenated.
+
+## Installation
+
```
Comment (balteravishay) [inline presidio-anonymizer/README.MD:27]: i think as a package should come first.

File: presidio-anonymizer/README.MD [hunk 4]
```diff
@@ -1 +1,121 @@
...
+### As service:
+
+Run presidio_anonymizer/app.py or docker container and use endpoint:
```
Comment (balteravishay) [inline presidio-anonymizer/README.MD:30]: this should either be more specific or link to another doc which is.

**Review by balteravishay — COMMENTED** [review]
some comments

File: presidio-anonymizer/README.MD (hunk 4 above)
Comment (shiranr) [reply presidio-anonymizer/README.MD:30]: I think it will be more specific once the dockerization of the package will be done WDYT?

Comment (balteravishay) [inline presidio-anonymizer/README.MD:30]: maybe run it with flask instead of docker?

Comment (balteravishay) [reply presidio-anonymizer/README.MD:30]: so maybe write how to run it with flask which is doable at this point

File: presidio-anonymizer/README.MD [hunk 5]
```diff
@@ -1 +1,121 @@
...
+POST /anonymize
+```
+
+[comment]: <shiranr> (TODO add result example when anonymizer is ready 2626)
+
+```
+RESULT
+```
+
+```json
```
[the remark this answers is not in the document]
Comment (shiranr) [reply presidio-anonymizer/README.MD:42]: An example will be added for the response once I am done. I have a todo + ticket on it.

File: presidio-anonymizer/README.MD [hunk 6]
```diff
@@ -1 +1,121 @@
...
+
+Presidio anonymizer comes by default with the following anonymizers:
+
+- Replace - replaces PII with desired value
+- Redact - removes PII completely from text
+- Hash - hash the PII using sha256
+- FPE - using ff1 algorithm for formatting-Preserving Encryption on the PII
+- Mask - mask the selected amount of characters of the PII with a given character
+
+Since text might have several PIIs, the anonymizer handle those scenarios in the
```
[the remark this answers is not in the document]
Comment (shiranr) [reply presidio-anonymizer/README.MD:16]: But we are not :) the first paragraph states a single PII was found.

File: presidio-anonymizer/README.MD (hunk 2 above)
Comment (shiranr) [reply presidio-anonymizer/README.MD:5]: Done

File: presidio-anonymizer/README.MD (hunk 4 above)
Comment (shiranr) [reply presidio-anonymizer/README.MD:30]: Done

File: presidio-anonymizer/README.MD [hunk 7]
```diff
@@ -1 +1,121 @@
...
+```
+
+```json
+{
+  "text": "hello world, my name is Jane Doe. My number is: 034453334",
+  "transformations": [
+    {
+      "DEFAULT": {
+        "type": "replace",
+        "new_value": "val"
```
[the remark this answers is not in the document]
Comment (shiranr) [reply presidio-anonymizer/README.MD:49]: Done

[the remark this answers is not in the document]
Comment (shiranr) [reply presidio-anonymizer/README.MD:49]: Although, I see in the proto there is an option to enter new value.

File: presidio-anonymizer/README.MD [hunk 8]
```diff
@@ -1 +1,126 @@
-# TODO add the document Omri wrote.
\ No newline at end of file
+# Presidio anonymizer
+
+## Description
+
+The Presidio anonymizer is a Python based module for anonymizing detected PII text
+entities with desired values.
+
+Presidio anonymizer comes by default with the following anonymizers:
```
Comment (navalev) [inline presidio-anonymizer/README.MD:8]: replace anonimizers with "transformations" @omri374 WDYT?

**Review by navalev — CHANGES_REQUESTED** [review]

File: presidio-anonymizer/README.MD (hunk 8 above)
Comment (omri374) [reply presidio-anonymizer/README.MD:8]: I don't have a strong preference here :) Maybe just for the sake of having it similar to the analyzer where we have the analyzer and different recognizers.

File: presidio-anonymizer/README.MD (hunk 6 above)
[the remark this answers is not in the document]
Comment (omri374) [reply presidio-anonymizer/README.MD:25]: This is that part which isn't clear IMHO. The sentence "since text might have several PIIs" is ambiguous.

File: presidio-anonymizer/README.MD (hunk 5 above)
[the remark this answers is not in the document]
Comment (omri374) [reply presidio-anonymizer/README.MD:84]: Cool

**Review by balteravishay — APPROVED** [review]

File: presidio-anonymizer/README.MD (hunk 6 above)
[the remark this answers is not in the document]
Comment (shiranr) [reply presidio-anonymizer/README.MD]: @omri374 I've changed the text. please check if it's ok?

File: presidio-anonymizer/README.MD (hunk 8 above)
Comment (navalev) [reply presidio-anonymizer/README.MD:8]: ok, lets close for now

**Review by navalev — APPROVED** [review]

File: presidio-anonymizer/README.MD [hunk 9]
```diff
@@ -1 +1,126 @@
...
+[comment]: <shiranr> (#TODO need to fill this once the the functionality is done 2656 - 2659)
+Each anonymizer requires different values:
+
+- Replace -
+- Redact - None
+- Hash - None
+- FPE -
+- Mask -
+
+Text might have zero to many PIIs, the anonymizer handle those scenarios in the following
```
Comment (omri374) [inline presidio-anonymizer/README.MD:25]:
```suggestion
As the input text could have potentially have overlapping PII entities, there are different anonymization scenarios:
```

Comment (omri374) [reply presidio-anonymizer/README.MD:25]: nit: Trying to rephrase to focus this on overlapping PII. Feel free to rephrase.

File: presidio-anonymizer/README.MD [hunk 10]
```diff
@@ -1 +1,126 @@
...
+ways:
+
+- One PII - uses a given or default transformation to anonymize and replace the PII text
```
Comment (omri374) [inline presidio-anonymizer/README.MD:28]: Maybe rephrase to "No overlap"? It could have multiple PII entities but they are not overlapping

File: presidio-anonymizer/README.MD [hunk 11]
```diff
@@ -1 +1,126 @@
...
+  entity.
+- Full overlap of PIIs - When one text have several PIIs, the PII with the higher score
+  will be taken. Between PIIs with similar scores, the selection will be arbitrary.
```
Comment (omri374) [inline presidio-anonymizer/README.MD:31]:
```suggestion
  will be taken. Between PIIs with identical scores, the selection will be arbitrary.
```

File: presidio-anonymizer/README.MD [hunk 12]
```diff
@@ -1 +1,126 @@
...
+- One PII is contained in another - anonymizer will use the PII with larger text.
+- Partial intersection - both will be returned concatenated.
```
Comment (omri374) [inline presidio-anonymizer/README.MD:33]: Consider providing examples of input and how the output would look like (say for replace)

File: presidio-anonymizer/README.MD [hunk 13]
```diff
@@ -1 +1,126 @@
...
+In folder presidio/presidio-anonymizer run:
+
+```
+pipenv sync
+```
+
+Start the server with flask (***this is a test server please do not use in prod***):
+
+```
+python app.py
```
Comment (omri374) [inline presidio-anonymizer/README.MD:67]: If we want them to run it with `pipenv` then the everything needs to run within pipenv. 
```suggestion
pipenv run app.py
```

File: presidio-anonymizer/README.MD (hunk 10 above)
Comment (shiranr) [reply presidio-anonymizer/README.MD:28]: DONE

File: presidio-anonymizer/README.MD (hunk 9 above)
Comment (shiranr) [reply presidio-anonymizer/README.MD]: DONE

File: presidio-anonymizer/README.MD (hunk 13 above)
Comment (shiranr) [reply presidio-anonymizer/README.MD:67]: DONE

File: presidio-anonymizer/README.MD (hunk 12 above)
Comment (shiranr) [reply presidio-anonymizer/README.MD:33]: DONE. @omri374 please re-review. I would like to push this.

**Review by navalev — DISMISSED** [review]

**Review by balteravishay — APPROVED** [review]

**Review by navalev — APPROVED** [review]
